galley/pkg/runtime/projections/serviceentry/convert/convert.go

```diff
@@ -106,16 +110,32 @@ func convertExportTo(annotations resource.Annotations) []string {
 
 // Endpoints converts k8s Endpoints to networking.ServiceEntry_Endpoint resources and extracts the service accounts for the endpoints.
 // The target ServiceEntry is passed as an argument (out) in order to enable object reuse in the future.
-func Endpoints(endpoints *coreV1.Endpoints, pods pod.Cache, out *networking.ServiceEntry) {
+func Endpoints(endpoints *coreV1.Endpoints, pods pod.Cache, out *networking.ServiceEntry) (notReadyEndpoints string) {
```

Comment:
Nit: seems a little confusing that `Endpoints` returns `notReadyEndpoints string`.

Reply:
yeah, agreed. That's where the processing really needs to be, unfortunately. Otherwise, we'd have to do it in `Annotations`, and we'd end up with redundant endpoint processing logic.